invenio_records_rest/facets.py

```diff
@@ -159,10 +168,11 @@ def default_facets_factory(search, index):
 
         # Query filter
         search, urlkwargs = _query_filter(
-            search, urlkwargs, facets.get("filters", {}))
+            search, urlkwargs, facets.get('filters', {}))
 
         # Post filter
         search, urlkwargs = _post_filter(
-            search, urlkwargs, facets.get("post_filters", {}))
+            search, urlkwargs, facets.get('post_filters', {}))
+        urlkwargs.update({'facets': selected_facets})
```

Comment:
```suggestion
        if selected_facets:
            urlkwargs.add('facets', ','.join(selected_facets))
```

The `facets` have to be set in the same format as it is received so that it can be parsed again into a list.